gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2

```diff
@@ -55,6 +55,47 @@ except ImportError:  # pragma: NO COVER
 {% endif %}{# service_version #}
 {% endmacro %}
 
+{% macro create_metadata(method) %}
+    {% if method.explicit_routing %}
```

Comment:
To resolve https://github.com/googleapis/gapic-generator-python/issues/2078, we'll probably need to add an extra check here for passed-in routing metadata:

```
metadata = tuple(metadata)
# add routing headers if not passed in petadata
if not any (m[0] == gapic_v1.routing_header.ROUTING_METADATA_KEY for m in metadata):
   {% if method.explicit_routing %}
   ...
```

I have https://github.com/googleapis/gapic-generator-python/pull/2079 open to try to address this, and I plan to fix it up to match this new structure after this PR is merged. But I'm mentioning it now in case you think it makes more sense to just address these together here

Reply:
For #2078 we need to update AIPs first so there is consistent behavior across all language-specific generators: https://google.aip.dev/client-libraries/4222

I'm going to go ahead without #2078 and there is still work pending